mentoring/mentoring.py

```diff
@@ -25,70 +22,275 @@
+        # Migrate stored data if necessary
+        self.migrate_fields()
 
-        correct_icon_url = self.runtime.local_resource_url(self, 'public/img/correct-icon.png')
+        fragment = Fragment()
+        child_content = u""
 
-        fragment.add_content(render_template('templates/html/mentoring.html', {
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if isinstance(child, MentoringMessageBlock):
+                pass
+            else:
+                try:
+                    if self.is_assessment and isinstance(child, StepMixin):
+                        child_fragment = child.render('assessment_step_view', context)
+                    else:
+                        child_fragment = child.render('mentoring_view', context)
+                except NoSuchViewError:
+                    if child.scope_ids.block_type == 'html' and getattr(self.runtime, 'is_author_mode', False):
+                        # html block doesn't support mentoring_view, and if we use student_view Studio will wrap
+                        # it in HTML that we don't want in the preview. So just render its HTML directly:
+                        child_fragment = Fragment(child.data)
+                    else:
+                        child_fragment = child.render('student_view', context)
+                fragment.add_frag_resources(child_fragment)
+                child_content += child_fragment.content
+
+        fragment.add_content(loader.render_template('templates/html/mentoring.html', {
             'self': self,
-            'named_children': named_children,
+            'title': self.display_name,
+            'child_content': child_content,
             'missing_dependency_url': self.has_missing_dependency and self.next_step_url,
         }))
         fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/mentoring.css'))
-        fragment.add_javascript_url(
-                    self.runtime.local_resource_url(self, 'public/js/vendor/underscore-min.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/vendor/underscore-min.js'))
+        js_file = 'public/js/mentoring_{}_view.js'.format('assessment' if self.is_assessment else 'standard')
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, js_file))
         fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/mentoring.js'))
-        fragment.add_resource(load_resource('templates/html/mentoring_progress.html').format(
-                completed=correct_icon_url),
-            "text/html")
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_attempts.html'), "text/html")
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_grade.html'), "text/html")
+
+        self.include_theme_files(fragment)
+        # Workbench doesn't have font awesome, so add it:
+        try:
```

Comment:
@nedbat  Number 1) I can easily address; that's a good point.

I'm not clear on what you mean by point 2), sorry. This code is already adding a URL for font awesome to the fragment. If you're saying to add the URL unconditionally, that would waste an HTTP request or two on every page load and would result in two (probably different) versions of font awesome being loaded with each page view. If you're saying to import the font awesome files and store them in this repo, that adds more files and complexity, and still has the duplicates issue. I'm unclear on what the advantage of that would be. Studio/LMS already has FA and we only add it here for the sake of unit tests and development of this XBlock in workbench/sdk.